File: src/main/com/lec12/Lec12Main.java
package main.com.lec12;

public class Lec12Main {
    public static void main(String[] args) {
        Person.Factory.newBaby("ABC");
        Person.newBaby("ABC");

        moveSomething(new Moveable() {
            @Override
            public void move() {
                System.out.println("움직인당~~");
            }

            @Override
            public void fly() {
                System.out.println("난당~~~");
            }
        });
    }

    private static void moveSomething(Moveable moveable) {
        moveable.move();
        moveable.fly();
    }
}

File: src/main/com/lec12/Lec12Main.kt
package main.com.lec12

fun main() {
    moveSomeThing(object: Moveable {
        override fun move() {
            println("움직인다아앙")
        }

        override fun fly() {
            println("난다아앙")
        }
    })
}

private fun moveSomeThing(moveable: Moveable) {
    moveable.move()
    moveable.fly()
}